vllm/v1/structured_output/utils.py

```diff
@@ -45,8 +45,7 @@ def check_object(obj: dict[str, Any]) -> bool:
             return True
 
         # Unsupported keywords for strings
-        if obj.get("type") == "string" and any(
```

Comment:
Please add some unit tests, similar to [[PR #15594](https://github.com/vllm-project/vllm/pull/15594)].